src/Razor/src/Microsoft.VisualStudio.LanguageServerClient.Razor/HtmlCSharp/RazorDiagnosticsResponse.cs

```diff
@@ -13,13 +11,15 @@ namespace Microsoft.VisualStudio.LanguageServerClient.Razor.HtmlCSharp
     // Note: This type should be kept in sync with the one in Razor.LanguageServer assembly.
     internal class RazorDiagnosticsResponse : IEquatable<RazorDiagnosticsResponse>
     {
-        public Diagnostic[] Diagnostics { get; set; }
+        public Diagnostic[]? Diagnostics { get; init; }
 
-        public int? HostDocumentVersion { get; set; }
+        public int? HostDocumentVersion { get; init; }
 
-        public bool Equals(RazorDiagnosticsResponse other)
+        public bool Equals(RazorDiagnosticsResponse? other)
         {
-            return Enumerable.SequenceEqual(Diagnostics, other.Diagnostics) &&
+            return
+                other is not null &&
+                Diagnostics.SequenceEqual(other.Diagnostics) &&
```

Comment:
Did you mean to leave this as `Enumerable.SequenceEqual(Diagnostics, other.Diagnostics)`, since `Diagnostics` is marked as nullable above?

Reply:
`Enumerable.SequenceEqual(Diagnostics, other.Diagnostics)` and `Diagnostics.SequenceEqual(other.Diagnostics)` are absolutelly the same in their behaviour. The latter is just shorter, so I've chosen it. There are no nullability warnings etc., so I don't see a problem here

Reply:
> There are no nullability warnings etc., so I don't see a problem here

`Diagnostics` is a `Diagnostics[]?` and there isn't a check here that proves that it's non-null in this location. So, from reading the code, it looks like it would be a problem. You're not seeing a nullability warning because `SequenceEquals` is an extension method and it's defined in the .NET Framework, so it isn't null-annotated.

Reply:
Extension methods are lowered to static calls anyway ([proof](https://sharplab.io/#v2:D4AQDABCCMAsDcBYAUOK0B0AZAlgOwEclkUQBmKAJggGEIBvFCZqCkWCAWQAoBKJlo2QsREAG4BDAE4Q8AVwC2AIwCmUgM4QAvLJUB3CLnUAXADz5jAPj4DRDW3ZbQANBDKvYrgKyuAbK4B2VwAOVwBOBxEAX2JIlhgyczwrWUVVDWodeQAbbOJHCDjmeWU1dUoMAGUVAjkVPABjFQBRWols7hL09V58liiUKKA=)), so if there is a nullability issue it doesn't matter in what form the code is written. However I agree, that having code like `<potential null>.ExtensionMethod()` is misleading, so I'll improve this code

Reply:
> Extension methods are lowered to static calls anyway ([proof](https://sharplab.io/#v2:D4AQDABCCMAsDcBYAUOK0B0AZAlgOwEclkUQBmKAJggGEIBvFCZqCkWCAWQAoBKJlo2QsREAG4BDAE4Q8AVwC2AIwCmUgM4QAvLJUB3CLnUAXADz5jAPj4DRDW3ZbQANBDKvYrgKyuAbK4B2VwAOVwBOBxEAX2JIlhgyczwrWUVVDWodeQAbbOJHCDjmeWU1dUoMAGUVAjkVPABjFQBRWols7hL09V58liiUKKA=)), so if there is a nullability issue it doesn't matter in what form the code is written. However I agree, that having code like `<potential null>.ExtensionMethod()` is misleading, so I'll improve this code

Thank you!